cicd/base/app/base.yaml

```diff
@@ -0,0 +1,357 @@
+        - !Sub ${pDefaultLogoutUrl}
+      PreventUserExistenceErrors: ENABLED
+      RefreshTokenValidity: 30
+      SupportedIdentityProviders:
+        - COGNITO
+        - Facebook
+        - Google
+        - AAF
+        - SignInWithApple
+      TokenValidityUnits:
+        AccessToken: minutes
+        IdToken: minutes
+        RefreshToken: days
+      UserPoolId: !ImportValue
+        Fn::Sub: ${pCognitoStackName}-UserPoolId
+
+  AppClientServerToServer:
+    Type: AWS::Cognito::UserPoolClient
+    Properties:
+      AccessTokenValidity: 60
+      AllowedOAuthFlows:
+        - client_credentials
+      AllowedOAuthFlowsUserPoolClient: True
+      AllowedOAuthScopes:
+        - ala/internal
+        - ala/attrs
+        - users/read
+      AuthSessionValidity: 3
+      CallbackURLs:
+        - !Sub "https://${pAppClientDomain}/callback?client_name=OidcClient"
+      ClientName: !Sub
+        - ${pProductName}-${ResourceName}-server-to-server
+        - ResourceName: !If [ IsDev, !Ref pCleanBranch, !Ref pEnvironment ]
+      DefaultRedirectURI: !Sub "https://${pAppClientDomain}/callback?client_name=OidcClient"
+      EnablePropagateAdditionalUserContextData: True
+      EnableTokenRevocation: True
+      ExplicitAuthFlows:
+        - ALLOW_ADMIN_USER_PASSWORD_AUTH
+        - ALLOW_CUSTOM_AUTH
+        - ALLOW_REFRESH_TOKEN_AUTH
+        - ALLOW_USER_SRP_AUTH
+      GenerateSecret: True
+      IdTokenValidity: 60
+      LogoutURLs:
+        - !Sub "https://${pAppClientDomain}"
+      PreventUserExistenceErrors: ENABLED
+      RefreshTokenValidity: 30
+      SupportedIdentityProviders:
+        - COGNITO
+        - Facebook
+        - Google
+        - AAF
+        - SignInWithApple
+      TokenValidityUnits:
+        AccessToken: minutes
+        IdToken: minutes
+        RefreshToken: days
+      UserPoolId: !ImportValue
+        Fn::Sub: ${pCognitoStackName}-UserPoolId
+
+
+Outputs:
+
+  UserdetailsRepositoryArn:
+    Description: The ECR repository ARN for the userdetails service
+    Value: !GetAtt UserdetailsRepository.Arn
+
+  UserdetailsRepositoryName:
+    Description: The ECR repository ARN for the userdetails service
+    Value: !Ref UserdetailsRepository
+
+  UserdetailsRepositoryUri:
+    Description: The ECR repository ARN for the userdetails service
+    Value: !GetAtt UserdetailsRepository.RepositoryUri
+    Export:
+      Name: !Sub ${AWS::StackName}-UserdetailsRepositoryUri
+
+  UserdetailsSecretArn:
+    Description: The Secrets name for userdetails
+    Value: !Ref UserdetailsSecret
+    Export:
+      Name: !Sub ${AWS::StackName}-UserdetailsSecretArn
+
+  UserdetailsRoleArn:
+    Description: The name of the userdetails role
+    Value: !GetAtt UserdetailsRole.Arn
+    Export:
+      Name: !Sub ${AWS::StackName}-UserdetailsRoleArn
+
+  UserPoolId:
+    Description: The Cognito User Pool ID
+    Value: !ImportValue
+      Fn::Sub: ${pCognitoStackName}-UserPoolId
```

Comment:
This is outputing an export value from another stack, whatever is using it should reference the cognito stack directly then this can be removed

Reply:
Fixed